### DISCUSSION TITLE
Iterating over multiple files in one record

### DISCUSSION THREAD

**Discussion by samfcheng** [pocketbase/pocketbase#1998] [Q&A] [answered]
I have a record with an images field
![image](https://user-images.githubusercontent.com/51756184/223506597-5b32c13c-6a2e-4695-8b64-45253db6a641.png)
I want to iterate over each of the images and show them.
Here's my current code:
```
<div class="about-images">
  {#each data.images as image}
    <img src={pb.getFileUrl(data.images, image)}/>
  {/each}
</div>
```
It seems to not be iteratable though. I tried getting an image url using `getFileUrl()`
```
pb.getFileUrl(data, data.images)
```
But that returns all of the images at once
```
api/files/yrgeyltz559l2dk/3ydmc87hwzzp2ue/r_9_drU5L4jwcM.jpeg%2Cr_8_gT8x0KMv2W.jpeg%2Cr_7_dVjB280dN1.jpeg
```
How can I  make this work?

**Comment by ganigeorgiev** [comment, marked as answer]
The first argument of `pb.getFileUrl` must be the record, and the second is the filename. Or in other words, it should be something like:

```svelte
<div class="about-images">
  {#each data.images as image}
    <img src={pb.getFileUrl(data, image)}/>
  {/each}
</div>
```

**Reply by samfcheng** [reply]
Doing that creates an error:
`Error: {#each} only iterates over array-like objects`
Just doing
`{pb.getFileUrl(data, data.images)}`
Results in a response of all three images:
api/files/yrgeyltz559l2dk/3ydmc87hwzzp2ue/r_9_drU5L4jwcM.***jpeg***%2Cr_8_gT8x0KMv2W.***jpeg***%2Cr_7_dVjB280dN1.***jpeg***

**Reply by ganigeorgiev** [reply]
I don't understand what do you mean. Could you provide a more complete code sample how `data` is loaded (or better a minimal reproducible repo)?

It is possible that if `data` comes from an async operation to not be loaded yet, aka. `data.images` to be `undefined` on pending request.

**Reply by samfcheng** [reply]
Data is loaded like this:
```
let data = {};
  onMount(async () => {
    data = await pb.collection('about_page').getOne('3ydmc87hwzzp2ue');
  });
```
Data is a record that has a field called images.
This images field has multiple files that are uploaded to it.
I want to iterate over this field and show all of the images that are uploaded to this file field.

A better way to do this might be to make a separate collection to make it easier to iterate over, but this is cleaner and more simple
Thanks for your help

**Reply by ganigeorgiev** [reply]
@Samalander0 The code:

```js
  {#each data.images as image}
    <img src={pb.getFileUrl(data, image)}/>
  {/each}
```

Should work if `data.images` is an array, but again because it is an async operation, while the record is being fetched it could be `undefined`. Alternatively, you can try something like:

```js
{#each (data.images || []) as image}
    <img src={pb.getFileUrl(data, image)} />
{/each}
```

If the above doesn't help, **please provide a more complete code sample** (or better a minimal reproducible repo).

**Reply by samfcheng** [reply]
That works, thanks so much!
